src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1269,8 +1271,8 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                     }
 
                     currentUri = new Uri(request.RequestUri, response.Headers.Location);
+
                     // Continue to handle redirection
-                    using (client = GetHttpClient(handleRedirect))
```

Comment:
Good step for https://github.com/PowerShell/PowerShell/issues/12764#issuecomment-1418557750 You could continue...

Reply:
I will try to continue, can you give me a more detailed to-do list?

Reply:
I laid out there already. The goal is to cache HttpClient/HttpClientHandler in WebSession. These HttpClient/HttpClientHandler cannot be changed after they start. So if a cmdlet is called with modified arguments when there is a WebSession, we have to recreate them. This makes me think that our first step is to move all initialization to BeginProcessing. If it's possible, we can then implement caching.